Password protected PDFs should emit password-related navigation delegate methods

hortont424 opened the pull request:
#### 3fec0ecd58593da08f46f9e3da99739da19e0618
<pre>
Password protected PDFs should emit password-related navigation delegate methods
<a href="https://bugs.webkit.org/show_bug.cgi?id=257364">https://bugs.webkit.org/show_bug.cgi?id=257364</a>
rdar://109858534

Reviewed by Megan Gardner, Wenson Hsieh and Aditya Keerthi.

* Source/WebKit/UIProcess/API/ios/WKWebViewIOS.h:
* Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm:
(-[WKWebView _showPasswordViewWithDocumentName:passwordHandler:]):
(-[WKWebView _hidePasswordView]):
(-[WKWebView _didRequestPasswordForDocument]):
(-[WKWebView _didStopRequestingPasswordForDocument]):
* Source/WebKit/UIProcess/Cocoa/NavigationState.h:
* Source/WebKit/UIProcess/Cocoa/NavigationState.mm:
(WebKit::NavigationState::setNavigationDelegate):
* Source/WebKit/UIProcess/ios/PageClientImplIOS.mm:
(WebKit::PageClientImpl::requestPasswordForQuickLookDocument):
Make the password-related navigation delegates exist on all IOS_FAMILY, not just
where USE(QUICKLOOK) is true, and adopt them for WKPDFView as well. Despite
the names being QuickLook-specific, clients generally want to do the same things
when a password field is up regardless of whether it&apos;s up for PDF or QuickLook
reasons.

* Source/WebKit/UIProcess/ios/WKPDFView.mm:
(-[WKPDFView web_initWithFrame:webView:mimeType:]):
(-[WKPDFView updateBackgroundColor]):
Drive-by; update the background color when entering and exiting the password view.

(-[WKPDFView pdfHostViewControllerDocumentDidRequestPassword:]):
(-[WKPDFView pdfHostViewController:documentDidUnlockWithPassword:]):
Call the navigation delegate.

Canonical link: <a href="https://commits.webkit.org/264600@main">https://commits.webkit.org/264600@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8bc0b789efd2807d9d2a8d1cb4f4844f6d131024

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/8059 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/8344 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/8557 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/9711 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/8160 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/10330 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/8255 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/11022 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/8203 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/9290 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/7345 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/9835 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/6591 "Passed tests") | | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/14956 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/7714 "Passed tests") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/7497 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/10861 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/7970 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/6485 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/7283 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/1939 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/11492 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/7711 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aprotyas commented on Source/WebKit/UIProcess/ios/PageClientImplIOS.mm:
```diff
@@ -971,7 +971,6 @@ static UIInterfaceOrientationMask toUIInterfaceOrientationMask(WebCore::ScreenOr
     }
 
     [m_webView _showPasswordViewWithDocumentName:fileName passwordHandler:passwordHandler.get()];
-    NavigationState::fromWebPage(*m_webView.get()->_page).didRequestPasswordForQuickLookDocument();
```
Is a `_didRequestPasswordForDocument` call being made elsewhere?

hortont424 replied:
It moved into _showPasswordViewWithDocumentName

hortont424 committed:
Password protected PDFs should emit password-related navigation delegate methods
https://bugs.webkit.org/show_bug.cgi?id=257364
rdar://109858534

Reviewed by Megan Gardner, Wenson Hsieh and Aditya Keerthi.

* Source/WebKit/UIProcess/API/ios/WKWebViewIOS.h:
* Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm:
(-[WKWebView _showPasswordViewWithDocumentName:passwordHandler:]):
(-[WKWebView _hidePasswordView]):
(-[WKWebView _didRequestPasswordForDocument]):
(-[WKWebView _didStopRequestingPasswordForDocument]):
* Source/WebKit/UIProcess/Cocoa/NavigationState.h:
* Source/WebKit/UIProcess/Cocoa/NavigationState.mm:
(WebKit::NavigationState::setNavigationDelegate):
* Source/WebKit/UIProcess/ios/PageClientImplIOS.mm:
(WebKit::PageClientImpl::requestPasswordForQuickLookDocument):
Make the password-related navigation delegates exist on all IOS_FAMILY, not just
where USE(QUICKLOOK) is true, and adopt them for WKPDFView as well. Despite
the names being QuickLook-specific, clients generally want to do the same things
when a password field is up regardless of whether it's up for PDF or QuickLook
reasons.

* Source/WebKit/UIProcess/ios/WKPDFView.mm:
(-[WKPDFView web_initWithFrame:webView:mimeType:]):
(-[WKPDFView updateBackgroundColor]):
Drive-by; update the background color when entering and exiting the password view.

(-[WKPDFView pdfHostViewControllerDocumentDidRequestPassword:]):
(-[WKPDFView pdfHostViewController:documentDidUnlockWithPassword:]):
Call the navigation delegate.

Canonical link: https://commits.webkit.org/264600@main
